src/ImpactX.cpp

```diff
@@ -59,6 +62,8 @@ namespace impactx
         //   before we start the evolve loop, we are in "step 0" (initial state)
         int global_step = 0;
 
+        bool early_params_checked = false; // check typos in inputs after step 1
```

Comment:
Reset the variable here, make it otherwise a member variable.
pls move comment as its docstring
```suggestion
        early_params_checked = false; // check typos in inputs after step 1
```